shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -814,9 +814,14 @@ - (void)viewDidAppear:(BOOL)animated {
   if ([_engine.get() viewController] == self) {
     [self onUserSettingsChanged:nil];
     [self onAccessibilityStatusChanged:nil];
+
+#if !APPLICATION_EXTENSION_API_ONLY
     if (UIApplication.sharedApplication.applicationState == UIApplicationStateActive) {
+#endif
       [[_engine.get() lifecycleChannel] sendMessage:@"AppLifecycleState.resumed"];
```

Comment:
Is it possible to end up sending this if the whole process hosting the extension itself was in the background, thus UIApplication.sharedApplication.applicationState != UIApplicationStateActive if we had access to the sharedApplication? If we sent in "resumed" without checking, might be end up in an engine state where we continue to issue GPU commands while in the background and crash?

Reply:
Seems like a race that I bet could happen in the app extension https://github.com/flutter/engine/pull/23175